navapbc/labs-decision-support-tool
# feat: Use URL query 'engine' parameter to set chatbot's configuration

## Ticket

https://navalabs.atlassian.net/browse/DST-327


## Changes


- Add `ChatEngineInterface` abstract class that acts as an interface for chat engines
- Refactor existing chatbot backend to `GuruMultiprogramEngine` class
- Add placeholder engines: `GuruSnapEngine` and `PolicyMichiganEngine`


## Testing
* Go to `chat/` or `/chat/?engine=unknownEngine`, expect "Available engines: ['guru-snap', 'policy-mi', 'guru-multiprogram']"
* Go to `/chat/?engine=guru-multiprogram`, expect "Chat engine started: Guru Multi-program Chat Engine"
   - Expect normal behavior after submitting a question
* Go to `/chat/?engine=guru-snap`, expect "Chat engine started: Guru SNAP Chat Engine"
   - Expect "TEMP: Replace with generated response once chunks are correct"
* Go to `/chat/?engine=policy-mi`, expect "Chat engine started: Michigan Bridges Policy Manual Chat Engine"
   - Expect "TEMP: Replace with generated response once chunks are correct"
